app/src/main/java/com/example/udefine/widgetManager.java

```diff
@@ -120,6 +120,65 @@ public void generate(ArrayList<Integer> componentList, ArrayList<String> title)
         }
     }
 
+    public ArrayList<String> getTitleTimeTag() {
+        int id, counter = 0;
+        String content = "";
+        ArrayList<String> titleTimeTag = new ArrayList<String>();
+        Iterator<HashMap.Entry<Integer, String>> iterator =
+                title_id_list.entrySet().iterator();
+
+        while (iterator.hasNext() && counter <= 2) {
+            // get Title and View
+            HashMap.Entry<Integer, String> entry = iterator.next();
+            id = entry.getKey();
+            View v = parentLinearLayout.findViewById(id);
+
+            if (v instanceof EditText) {
+                // get Title
+                EditText e = (EditText) v;
+                content = e.getText().toString();
+            } else if (v instanceof Button) {
+                // get date + time
+                Button b = (Button) v;
+                String tmp = b.getText().toString();
+
+                // if Date and Time is not set, set content to null
+                if (tmp.equals("Date")) {
```

Comment:
What does this if section do?
Why the text should be "Date" or "Time"?

Reply:
If the user did not set the date or time, the text of buttons would be `Date` and `Time`,
thus we will need to set the value to null